### PR TITLE
Caliper modifier

### PR DESCRIPTION
**Pull request by rfhaque**
[[DO NOT MERGE]] This commit provides a draft implementation of the caliper modifier for benchpark

### CODE REVIEW

File: configs/ats4/auxiliary_software_files/packages.yaml [hunk 1]
```diff
@@ -1,6 +1,15 @@
 packages:
   all:
     require: target=x86_64
+  papi:
+    externals:
+    - spec: papi@7.0.0.2
+      prefix: /opt/cray/pe/papi/7.0.0.2
```
Comment (pearce8) [inline configs/ats4/auxiliary_software_files/packages.yaml:7]: @becker33 @scheibelp Should we be looking for a system install of PAPI, or is it advisable to build PAPI for each system?

File: configs/cts1/auxiliary_software_files/packages.yaml [hunk 2]
```diff
@@ -1,4 +1,13 @@
 packages:
+  papi:
+    externals:
+    - spec: papi@6.0.0.1
+      prefix: /usr/tce/packages/papi/papi-6.0.0.1
+      buildable: false
+  elfutils:
+    externals:
+    - spec: elfutils@0.188
```
Comment (pearce8) [inline configs/cts1/auxiliary_software_files/packages.yaml:9]: @becker33 @scheibelp Same question for elf

File: configs/cts1/auxiliary_software_files/packages.yaml [hunk 3]
```diff
@@ -23,6 +32,11 @@ packages:
     - spec: intel-oneapi-mkl@2022.1.0
```
Comment (pearce8) [inline configs/cts1/auxiliary_software_files/packages.yaml:32]: @rfhaque why do we need oneapi here?

File: experiments/amg2023/openmp/ramble.yaml [hunk 4]
```diff
@@ -36,8 +37,18 @@ ramble:
             omp_num_threads: '{threads_per_node_core} * {n_nodes}'
           experiments:
             amg2023_omp_{compilers}_problem1_{n_nodes}_{omp_num_threads}_{px}_{py}_{pz}_{nx}_{ny}_{nz}:
+              modifiers:
+                - name: caliper
+                  mode: spot 
+              success_criteria:
```
Comment (pearce8) [inline experiments/amg2023/openmp/ramble.yaml:43]: @rfhaque Do we have to have a success criteria, even if we use the output of this modifier only in post-processing?

File: configs/cts1/auxiliary_software_files/packages.yaml (hunk 3 above)
Comment (rfhaque) [reply configs/cts1/auxiliary_software_files/packages.yaml:32]: > @rfhaque why do we need oneapi here?

@pearce intel-oneapi-mkl is the blas and lapack implementation on cts1 machine

File: experiments/amg2023/openmp/ramble.yaml (hunk 4 above)
Comment (rfhaque) [reply experiments/amg2023/openmp/ramble.yaml:43]: > @rfhaque Do we have to have a success criteria, even if we use the output of this modifier only in post-processing?

@pearce8 We still need some way to assess the success/failure of the experiment using "ramble workspace analyze". We use the modifier output in post-processing to extract the FOM and use that in the success criteria.

**Comment by rfhaque** [comment]
@pearce8 Closing this PR as the top-down and cuda modifier implementations have been split into their own branches

**Pull request closed without merging**